js/repeater.js

```diff
@@ -691,90 +726,71 @@
 			}
 		},
 
-		runRenderer: function (viewTypeObj, data, callback) {
-			var $container, i, l, response, repeat, subset;
-
-			function addItem ($parent, resp) {
-				var action;
-				if (resp) {
-					action = (resp.action) ? resp.action : 'append';
-					if (action !== 'none' && resp.item !== undefined) {
-						$parent = (resp.container !== undefined) ? $(resp.container) : $parent;
-						$parent[action](resp.item);
-					}
-				}
-			}
-
+		renderItems: function renderItems (viewTypeObj, data, callback) {
 			if (!viewTypeObj.render) {
 				if (viewTypeObj.before) {
-					response = viewTypeObj.before.call(this, {
+					var addBefore = viewTypeObj.before.call(this, {
 						container: this.$canvas,
 						data: data
 					});
-					addItem(this.$canvas, response);
+					addItem(this.$canvas, addBefore);
 				}
 
-				$container = this.$canvas.find('[data-container="true"]:last');
+				var $container = this.$canvas.find('[data-container="true"]:last');
 				$container = ($container.length > 0) ? $container : this.$canvas;
 
 				if (viewTypeObj.renderItem) {
-					repeat = viewTypeObj.repeat || 'data.items';
+					var subset;
+					var repeat = viewTypeObj.repeat || 'data.items';
 					repeat = repeat.split('.');
 					if (repeat[0] === 'data' || repeat[0] === 'this') {
 						subset = (repeat[0] === 'this') ? this : data;
 						repeat.shift();
 					} else {
 						repeat = [];
 						subset = [];
-						if (window.console && window.console.warn) {
-							window.console.warn('WARNING: Repeater plugin "repeat" value must start with either "data" or "this"');
-						}
+						logWarn('WARNING: Repeater plugin "repeat" value must start with either "data" or "this"');
 					}
 
-					for (i = 0, l = repeat.length; i < l; i++) {
-						if (subset[repeat[i]] !== undefined){
-							subset = subset[repeat[i]];
+					for (var repeatI = 0, repeatL = repeat.length; repeatI < repeatL; repeatI++) {
+						if (subset[repeat[repeatI]] !== undefined) {
+							subset = subset[repeat[repeatI]];
 						} else {
 							subset = [];
-							if (window.console && window.console.warn) {
-								window.console.warn('WARNING: Repeater unable to find property to iterate renderItem on.');
-							}
-							break;
+							logWarn('WARNING: Repeater unable to find property to iterate renderItem on.');
+							repeatI = repeat.length;
 						}
 					}
 
-					for (i = 0, l = subset.length; i < l; i++) {
-						response = viewTypeObj.renderItem.call(this, {
+					for (var subItemI = 0, subsetL = subset.length; subItemI < subsetL; subItemI++) {
```

Comment:
Not completely sure but length may already by a cached value

http://stackoverflow.com/questions/5752906/is-reading-the-length-property-of-an-array-really-that-expensive-an-operation